backend/app/analysis/frequency.py

```diff
@@ -0,0 +1,87 @@
+from collections import Counter
+from ..models import (
+    Corpus,
+    Document,
+    Gender
+)
+
+
+def _get_gender_word_frequencies_relative(gender_word_counts):
+    """
+    A private helper function that examines identifier counts keyed to Gender instances,
+    determines the total count value of all identifiers across Gender instances,
+    and returns the percentage of each identifier count over the total count.
+
+    :param gender_word_counts: a dictionary keying gender instances to string identifiers keyed to
+                               integer counts.
+    :return: a dictionary with the integer counts transformed into float values representing
+             the identifier count as a percentage of the total identifier counts across all
+             identifiers.
+    """
+
+    output = {}
+    total_word_count = 0
+    for gender in gender_word_counts:
+        for word in gender_word_counts[gender]:
+            total_word_count += gender_word_counts[gender][word]
+
+    for gender in gender_word_counts:
+        output[gender] = {}
+        for word, original_count in gender_word_counts[gender].items():
+            try:
+                frequency = original_count / total_word_count
+            except ZeroDivisionError:
+                frequency = 0
+            output[gender][word] = frequency
+
+    return output
+
+
+def run_single_analysis(doc_obj, genders):
+    """
+    This method generates a dictionary that includes a Counter (count) that keys
+    Document instances to Gender instances to Counter instances representing the total
+    number of instances of each Gender's pronouns in a given Document, a dictionary (frequency)
+    keying Document instances to Gender instances to dictionaries of the shape {str:float}
+    representing the total number of instances of each Gender's pronouns over the total word count
+    of that Document; and a dictionary (relative) keying Document instances to Gender instances
+    to dictionaries of the shape {str:float} representing the relative percentage of Gender
+    pronouns across all Gender instances in a given Document instance.
+
+    :param doc_obj: an instance of the Document model
+    :param genders: a list of Gender objects
+    :return: a dictionary containing the frequency analyses of the Document instance
+    """
+    count = Counter()
+    frequency = {}
+
+    for gender in genders:
+        count[gender.label] = doc_obj.get_count_of_words(gender.pronouns)
+        frequency[gender.label] = doc_obj.get_word_freqs(gender.pronouns)
```

Comment:
Internal to the `frequency` module, we should probably allow the model instances themselves be the keys of these data structures rather than some field on them. We can let a serializer transform those into JSON-friendly values. Using the instances themselves might help with debugging down the road (i.e., we can put a breakpoint and investigate the instance itself rather than just a field on it).

Reply:
This is :awesome: and works well. I do think it'd be more useful down the road if we slightly change the dictionary keys to use the `PronounSeries` field names ('subj', 'obj', 'pos_det', 'pos_pro', 'reflex'). I'm imagining a scenario down the road where we want to know the relative frequency of all `Gender`'s subjects, which will be trickier with the current implementation than if we key value count/frequency to the key 'subj'.

Reply:
I'm slightly confused about which key we should change to `PronounSeries` field names. Will the structure be something like this: `{frequency: {Gender: {'subj': ...}}}`?